configuration/src/main/java/io/airlift/configuration/ConfigurationFactory.java

```diff
@@ -688,11 +687,6 @@ private static Object coerce(TypeToken<?> type, String value)
                     .collect(toImmutableList());
         }
 
-        if (LIST_OF_STRINGS_TYPE_TOKEN.isSubtypeOf(type)) {
```

Comment:
This can be part of `Support list and set injection into configuration` commit

Also, reword the commit and CHANGES entry `Support generic list and set injection ...` -- this because we had _some_ list injection, but now we have better one.